#!/usr/bin/env python3

from utils import Meta, HTTPException
from uuid import uuid4

with Meta() as meta:
  password = meta.create_user('foo@bar.baz')
  res = meta.post('login',
                  {
                    'device_id': str(uuid4()),
                    'email': 'foo@bar.baz',
                    'password': password,
                    'pick_trophonius': False,
                  })
  assert res['success']
  assert res['email'] == 'foo@bar.baz'
  res = meta.get('user/self')
  assert res['success']
  assert res['email'] == 'foo@bar.baz'
  assert res['creation_time'] is not None
  res = meta.post('logout')
  assert res['success']
  try:
    res = meta.get('user/self')
  except HTTPException as e:
    assert e.status == 403
